### ISSUE TITLE
Incorrect behavior of ENTER in XML documentation comment when whitespace is to the right of the editor caret

### ISSUE THREAD

**Issue by DustinCampbell** [dotnet/roslyn#2091]
**Repro Steps:**

* Enter the following code, and position the caret at the location indicated by `|`. *Note that the code block is indented.*
```C#
            /// <summary>
            /// foo| bar
            /// </summary>
            public void Foo() { }
```

* Press Enter

**Expected Result:**
```C#
        /// <summary>
        /// foo
        /// |bar
        /// </summary>
        public void Foo() { }
```

**Actual Result:**
```C#
        /// <summary>
        /// foo
       /// | bar
        /// </summary>
        public void Foo() { }
```

**Comment by sharwell** [comment]
Sorry I forgot to move this. :no_mouth:

**Comment by Pilchie** [comment]
Another minor irritation that we'll need to clean up in 1.1

**Comment by DustinCampbell** [comment]
Oops. Closed inadvertently.

**Comment by DustinCampbell** [comment]
Fixed with #4816.

**Comment by sharwell** [comment]
:bulb: @DustinCampbell You don't need to comment that it was fixed if your pull request contains "Fixed #2091" (which it did). GitHub automatically adds the link when the pull request is created, and then closes the issue and notifies any subscribers when the pull request is merged.

:thought_balloon: The one thing GitHub *doesn't* do is notify subscribers when the pull request is created.

**Comment by DustinCampbell** [comment]
@sharwell, I realize that. I just wanted to clarify since I'd incorrectly closed this issue earlier by referencing it in an unrelated PR.
